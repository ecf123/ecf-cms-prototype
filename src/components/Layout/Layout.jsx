import DropDown from "../DropDown/DropDown";
import "./Layout.scss";
import topicArrow from "../../assets/upside-down-arrow-grey.svg";
const Layout = ({ children }) => {
  return <div className="layout"><DropDown topicTitle="Business"lessonType={topicArrow} lessonTitle="Suits" lessonLink="Â£"/>{children}</div>;
};

export default Layout;
