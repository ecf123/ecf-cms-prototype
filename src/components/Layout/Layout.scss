@use "../../styles/variables" as *;

.layout {

  display: grid;
  grid-template-columns: 16.5rem auto;
  grid-template-rows: auto 1fr;

  &__nav-bar{
    grid-area: 1/1/3/2;
  }

  &__header{
    display: flex;
    justify-content: space-between;
    padding: 2rem;
    align-items: center;
    padding-right: 0;
    
    &--title{
      color: $color-blue-navy-dark;
      font-size: $font-size-extra-large;
      font-family: $font-secondary;
    }
  }
}
