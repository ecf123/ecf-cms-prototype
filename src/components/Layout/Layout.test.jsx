import userEvent from "@testing-library/user-event";
import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import Layout from "./Layout";

it("Should render layout with children", () => {
  const { container } = customRender(
    <Layout>
      <p>some content</p>
    </Layout>
  );

  expect(container).toMatchSnapshot();
});

it("should render the correct page title when clicking", () => {
  customRender( <Layout /> )

  const dashBoard = screen.queryAllByText("Dashboard")[0]
  userEvent.click(dashBoard)

  const pageTitle = screen.getByRole("heading", {level: 1})

  expect(pageTitle.innerHTML).toBe("Dashboard");

})


it("when clicking the marketplace, the title should be the same with the choice", () => {
  customRender( <Layout /> )

  const marketPlace = screen.queryAllByText("Marketplace")[0]
  userEvent.click(marketPlace)

  const pageTitle = screen.getByRole("heading", {level: 1})

  expect(pageTitle.innerHTML).toBe("Marketplace");

})
