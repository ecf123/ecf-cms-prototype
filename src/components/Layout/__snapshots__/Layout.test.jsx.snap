// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render layout with children 1`] = `
<div>
  <div
    class="layout"
  >
    <aside
      class="layout__nav-bar"
    >
      <div
        class="nav-bar"
      >
        <div
          class="nav-bar__container"
        >
          <div
            class="nav-bar__logo"
          >
            <a
              class="link"
              href="/*"
            >
              <div
                class="logo"
              >
                <div
                  class="logo__box"
                >
                  <img
                    alt="The Future Hub Logo"
                    class="logo__image"
                    src="ECFP-company-logo.svg"
                  />
                </div>
                <p
                  class="logo__top-title"
                >
                  THE
                </p>
                <h3
                  class="logo__title"
                >
                  FUTURE HUB
                </h3>
              </div>
            </a>
          </div>
          <ul
            class="nav-bar__link-list"
          >
            <a
              class="link"
              href="/"
            >
              <div
                class="nav-bar-item selected"
              >
                <div
                  class="nav-bar-item__image"
                >
                  <svg>
                    dashboard-windows-square.svg
                  </svg>
                </div>
                <p
                  class="nav-bar-item__text"
                >
                  Dashboard
                </p>
              </div>
            </a>
            <a
              class="link"
              href="/"
            >
              <div
                class="nav-bar-item"
              >
                <div
                  class="nav-bar-item__image"
                >
                  <svg>
                    pathways-chain-icon.svg
                  </svg>
                </div>
                <p
                  class="nav-bar-item__text"
                >
                  Pathways
                </p>
              </div>
            </a>
            <a
              class="link"
              href="/"
            >
              <div
                class="nav-bar-item"
              >
                <div
                  class="nav-bar-item__image"
                >
                  <svg>
                    article-book-icon.svg
                  </svg>
                </div>
                <p
                  class="nav-bar-item__text"
                >
                  Articles
                </p>
              </div>
            </a>
            <a
              class="link"
              href="/"
            >
              <div
                class="nav-bar-item"
              >
                <div
                  class="nav-bar-item__image"
                >
                  <svg>
                    marketplace-shopping-trolley-icon.svg
                  </svg>
                </div>
                <p
                  class="nav-bar-item__text"
                >
                  Marketplace
                </p>
              </div>
            </a>
            <a
              class="link"
              href="/"
            >
              <div
                class="nav-bar-item"
              >
                <div
                  class="nav-bar-item__image"
                >
                  <svg>
                    users-people-icon.svg
                  </svg>
                </div>
                <p
                  class="nav-bar-item__text"
                >
                  Users
                </p>
              </div>
            </a>
          </ul>
          <a
            class="link"
            href="/"
          >
            <div
              class="nav-bar-item"
            >
              <div
                class="nav-bar-item__image"
              >
                <svg>
                  cog-icon.svg
                </svg>
              </div>
              <p
                class="nav-bar-item__text"
              >
                Settings
              </p>
            </div>
          </a>
        </div>
      </div>
    </aside>
    <header
      class="layout__header"
    >
      <div
        class="header-container"
      >
        <div
          class="header-container__top"
        >
          <h1
            class="page-title page-title--small"
          >
            Dashboard
          </h1>
          <div
            class="header-profile"
          >
            <img
              alt="user avatar"
              class="header-profile__image"
              src="profile-picture.svg"
            />
            <div
              class="header-profile__text"
            >
              <h2
                class="header-profile__name"
              >
                Brett Zieme
              </h2>
              <p
                class="header-profile__role"
              >
                Admin
              </p>
            </div>
            <button
              class="header-profile__arrow"
            >
              <img
                alt="toggle"
                src="upside-down-arrow-grey.svg"
              />
            </button>
          </div>
        </div>
        <h1 />
      </div>
    </header>
    <main
      class="layout__body"
    >
      <p>
        some content
      </p>
    </main>
  </div>
</div>
`;
